deploy/examples/create-external-cluster-resources.py

```diff
@@ -826,7 +829,44 @@ def validate_rados_namespace(self):
             raise ExecutionFailureException(
                 ("The provided rados Namespace, '{}', is not found in the pool '{}'").format(
                     rados_namespace, rbd_pool_name))
-
+    
+    def get_rgw_fsid(self):
+        access_key = self.out_map['RGW_ADMIN_OPS_USER_ACCESS_KEY']
+        access_key = "WC2D69F9L7KEOJVETKER"
+        secret_key = self.out_map['RGW_ADMIN_OPS_USER_SECRET_KEY']
+        secret_key = "3iBk9hrYydYNd5QURzG8kXHo20muJXATPC2Xtpzf"
+        http_request = "GET"
+        rgw_endpoint = self._arg_parser.rgw_endpoint
+        resource = "/admin/info"
+        query_string = "format=json"
+        contentType = "application/x-compressed-tar"
+        dateTime = datetime.datetime.utcnow()
+        dateTimeFormat = dateTime.strftime('%a, %d %b %Y %X +0000')
+        headerToSign = http_request + "\n" + "\n" + contentType + "\n" + dateTimeFormat + "\n" + resource
+        print(headerToSign)
+        cmd = "echo -en " + headerToSign + " | openssl sha1 -hmac " + secret_key +" -binary | base64"
+        try:
+            signature_decode = subprocess.Popen(cmd,stderr=subprocess.PIPE,stdout=subprocess.PIPE,shell=True,universal_newlines=True)
+            signature = signature_decode.communicate()[0].strip()
+        except subprocess.CalledProcessError as execErr:
+            err_msg = "failed to execute command %s. Output: %s. Code: %s. Error: %s" % (
+                cmd, execErr.output, execErr.returncode, execErr.stderr)
+            raise Exception(err_msg)
+        request_url = "http://" + rgw_endpoint + resource + "?" + query_string
```

Comment:
what if RGW is running in htttps?

Reply:
@thotz I am not sure about it,
If rgw endpoint can be configured with both ways, how we can detect it,
Or we should ask from user about it as a cli parameter ?

Reply:
@parth-gr : AFAIR the script has parameter to TLS certs so that it will create k8s out of it. If it is passed then  we can assume it is `https` otherwise http